src/ProjectSystemTools/BuildLogging/Model/Build.cs

```diff
@@ -3,104 +3,42 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
-using Microsoft.VisualStudio.ProjectSystem.Tools.TableControl;
-using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model.Backend;
+using Microsoft.VisualStudio.ProjectSystem.Tools.Providers;
 
 namespace Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model
 {
+    // server side data (deals with log files)
     internal sealed class Build : IComparable<Build>, IDisposable
     {
-        public BuildType BuildType { get; }
-
-        public IEnumerable<string> Dimensions { get; }
-
-        public IEnumerable<string> Targets { get; }
-
-        public DateTime StartTime { get; }
-
-        public TimeSpan Elapsed { get; private set; }
-
-        public BuildStatus Status { get; private set; }
-
-        public string ProjectPath { get; }
-
+        // TODO: If needed, make an issue: implement more robust ID system (that doesn't use ints)
+        public static int SharedBuildID { get; private set; }
+        public BuildSummary BuildSummary { get; private set; }
         public string LogPath { get; private set; }
 
         public Build(string projectPath, IEnumerable<string> dimensions, IEnumerable<string> targets, BuildType buildType, DateTime startTime)
         {
-            ProjectPath = projectPath;
-            Dimensions = dimensions.ToArray();
-            Targets = targets?.ToArray() ?? Enumerable.Empty<string>();
-            BuildType = buildType;
-            StartTime = startTime;
-            Status = BuildStatus.Running;
+            BuildSummary = new BuildSummary(SharedBuildID, projectPath, dimensions, targets, buildType, startTime);
+            SharedBuildID++;
```

Comment:
If there's any chance that this code could be run in parallel from multiple threads, then using `++` is not safe and could produce duplicate Ids.

To ensure the sequence of Ids does not produce duplicates, you can use the `Interlocked` class. For example:


```suggestion
            int nextId = Interlocked.Increment(ref SharedBuildID);
            BuildSummary = new BuildSummary(nextId, projectPath, dimensions, targets, buildType, startTime);
```